RecoTracker/MkFit/plugins/MkFitOutputConverter.cc

```diff
@@ -0,0 +1,580 @@
+    }
+    LogTrace("MkFitOutputConverter") << "Setting DetLayer for index " << index << " subdet " << id.subdetId()
+                                     << " layer " << ttopo.layer(id) << " ptr " << layer;
+
+    dets[index] = layer;
+  };
+
+  // TODO: currently hardcoded...
+  // Logic copied from mkfit::LayerNumberConverter
+  unsigned int off = 1;
+  // BPix
+  set(0, ttopo.pxbDetId(1, 0, 0));
+  set(1, ttopo.pxbDetId(2, 0, 0));
+  set(2, ttopo.pxbDetId(3, 0, 0));
+  set(3, ttopo.pxbDetId(4, 0, 0));
+  // TIB
+  set(off + 3, ttopo.tibDetId(1, 0, 0, 0, 0, 0));
+  set(off + 4, ttopo.tibDetId(1, 0, 0, 0, 0, 1));
+  set(off + 5, ttopo.tibDetId(2, 0, 0, 0, 0, 0));
+  set(off + 6, ttopo.tibDetId(2, 0, 0, 0, 0, 1));
+  set(off + 7, ttopo.tibDetId(3, 0, 0, 0, 0, 0));
+  set(off + 8, ttopo.tibDetId(4, 0, 0, 0, 0, 0));
+  // TOB
+  set(off + 9, ttopo.tobDetId(1, 0, 0, 0, 0));
+  set(off + 10, ttopo.tobDetId(1, 0, 0, 0, 1));
+  set(off + 11, ttopo.tobDetId(2, 0, 0, 0, 0));
+  set(off + 12, ttopo.tobDetId(2, 0, 0, 0, 1));
+  set(off + 13, ttopo.tobDetId(3, 0, 0, 0, 0));
+  set(off + 14, ttopo.tobDetId(4, 0, 0, 0, 0));
+  set(off + 15, ttopo.tobDetId(5, 0, 0, 0, 0));
+  set(off + 16, ttopo.tobDetId(6, 0, 0, 0, 0));
+
+  auto setForward = [&](unsigned int side) {
+    // FPix
+    set(off + 0, ttopo.pxfDetId(side, 1, 0, 0, 0));
+    set(off + 1, ttopo.pxfDetId(side, 2, 0, 0, 0));
+    set(off + 2, ttopo.pxfDetId(side, 3, 0, 0, 0));
+    // TID+
+    off += 1;
+    set(off + 2, ttopo.tidDetId(side, 1, 0, 0, 0, 0));
+    set(off + 3, ttopo.tidDetId(side, 1, 0, 0, 0, 1));
+    set(off + 4, ttopo.tidDetId(side, 2, 0, 0, 0, 0));
+    set(off + 5, ttopo.tidDetId(side, 2, 0, 0, 0, 1));
+    set(off + 6, ttopo.tidDetId(side, 3, 0, 0, 0, 0));
+    set(off + 7, ttopo.tidDetId(side, 3, 0, 0, 0, 1));
+    // TEC
+    set(off + 8, ttopo.tecDetId(side, 1, 0, 0, 0, 0, 0));
+    set(off + 9, ttopo.tecDetId(side, 1, 0, 0, 0, 0, 1));
+    set(off + 10, ttopo.tecDetId(side, 2, 0, 0, 0, 0, 0));
+    set(off + 11, ttopo.tecDetId(side, 2, 0, 0, 0, 0, 1));
+    set(off + 12, ttopo.tecDetId(side, 3, 0, 0, 0, 0, 0));
+    set(off + 13, ttopo.tecDetId(side, 3, 0, 0, 0, 0, 1));
+    set(off + 14, ttopo.tecDetId(side, 4, 0, 0, 0, 0, 0));
+    set(off + 15, ttopo.tecDetId(side, 4, 0, 0, 0, 0, 1));
+    set(off + 16, ttopo.tecDetId(side, 5, 0, 0, 0, 0, 0));
+    set(off + 17, ttopo.tecDetId(side, 5, 0, 0, 0, 0, 1));
+    set(off + 18, ttopo.tecDetId(side, 6, 0, 0, 0, 0, 0));
+    set(off + 19, ttopo.tecDetId(side, 6, 0, 0, 0, 0, 1));
+    set(off + 20, ttopo.tecDetId(side, 7, 0, 0, 0, 0, 0));
+    set(off + 21, ttopo.tecDetId(side, 7, 0, 0, 0, 0, 1));
+    set(off + 22, ttopo.tecDetId(side, 8, 0, 0, 0, 0, 0));
+    set(off + 23, ttopo.tecDetId(side, 8, 0, 0, 0, 0, 1));
+    set(off + 24, ttopo.tecDetId(side, 9, 0, 0, 0, 0, 0));
+    set(off + 25, ttopo.tecDetId(side, 9, 0, 0, 0, 0, 1));
+  };
+
+  // plus
+  off = 17 + 1;
+  setForward(2);
+
+  // minus
+  off = 17 + 1 + 25 + 2;
```

Comment:
It would be nice to obtain some of these values from the geometry, or otherwise make them into named constants, but I'm not sure how easily that could be done in this case.

Reply:
Named constants should be possible.